presto-main/src/main/java/io/prestosql/operator/scalar/StringFunctions.java

```diff
@@ -726,6 +728,31 @@ public static Slice rightPad(@SqlType("varchar(x)") Slice text, @SqlType(Standar
         return pad(text, targetLength, padString, text.length());
     }
 
+    @Description("Repeats a string given number of times")
+    @ScalarFunction
+    @LiteralParameters("x")
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice replicate(@SqlType("varchar(x)") Slice inputSlice, @SqlType(StandardTypes.BIGINT) long times)
+    {
+        int repeatTimes = Ints.saturatedCast(times);
+        checkCondition(repeatTimes > 0, INVALID_FUNCTION_ARGUMENT, "Invalid number of string repetitions: " + times);
+
+        int length = inputSlice.length();
+        int size = length * repeatTimes;
+
+        try {
+            Slice buffer = Slices.allocate(size);
+            for (int i = 0; i < times; i++) {
+                buffer.setBytes(i * length, inputSlice, 0, length);
+            }
+
+            return buffer;
+        }
+        catch (SliceTooLargeException e) {
+            throw new PrestoException(EXCEEDED_FUNCTION_MEMORY_LIMIT, e.getMessage());
```

Comment:
We should cap the size at a more reasonable value. A 1GB allocation might work, for instance, but a page that big will cause all sorts of problems. I would pick a limit in the order of megabytes to align with the target size for a page. See PageBuilderStatus.DEFAULT_MAX_PAGE_SIZE_IN_BYTES